### DISCUSSION TITLE
WIP: Links to stuff that could interesting to others playing with jscad

### DISCUSSION THREAD

**Discussion by hrgdavor** [jscad/OpenJSCAD.org#1158] [Ideas]
# The distraction rabbit hole

## things I am working on (shameless plug)
- #1141 - multi-part project template
 - #944 - a long running process of making a new version of jscad web that I would be able to help maintain.
 - [instancing](https://3d.hrg.hr/jscad/three/threejscad2.html?uri=model.logos.js) performance to display large number of shapes that are same geometry but transformed.
 - [procedural screw](https://3d.hrg.hr/jscad/three/threejscad2.html?uri=#model.screw.js) example of performance benefits to procedurally generate common stuff aside from primitives (created some time ago, not yet integrated to jscad, likely will be a node module)
 - [screw dimensions lib](https://boltsparts.github.io/en/index.html) - providing a common database that can be utilized by many different CAD applications. Hopefully I can make a superfast generator for these based on link above.

## standards
 - [screw dimensions lib](https://boltsparts.github.io/en/index.html) - providing a common database that can be utilized by many different CAD applications _(Also mentioned in my personal section)_
 - 

## BOOLEANS
Other engines that do booleans

### Manifold (WASM)
A new boolean library that creates manifold shapes(no more errors in slicer) claims up-to 1000x performance gain over csg.js/(that is used by jscad). It is very promising and might be good to consider to offer as an option to use with jscad modeling. A quick test of cutting a sphere(256 segments) from a cube gives 35 sec in jscad, versus 0.07s in manifold. And jscad produces non-manifold 3mf.

- git: https://github.com/elalish/manifold
- editor: https://manifoldcad.org/

### Walnut (WASM)
An ineresting different approach to 3d booleans.
- https://github.com/bluelightning32/walnut
- https://github.com/bluelightning32/walnut-javascript - wasm

### Martinez
- repo - https://github.com/mfogel/polygon-clipping
- paper - https://github.com/mfogel/polygon-clipping/blob/HEAD/paper.pdf

### CGAL (not JS, no WASM)
booelans and everytinhg else and if u are like me, could be you are not even aware of it.
 - https://www.cgal.org/

### OpenSCAD
The inspiration that for the jscad, in the old days when WASM was not yet a thing
- https://github.com/DSchroer/openscad-wasm - WASM build
- https://openscad.cloud/ - online editor based on the WASM

### OCCT - OpenCascade
This one is different in that it uses BREP to describe geometry, but is used very much in desktop apps like FreeCAD.

- https://github.com/donalffons/opencascade.js - is WASM build of it
- https://zalo.github.io/CascadeStudio/ is one attempt in providing api
- https://replicad.xyz/ is another API on top of the WASM by author that did not like CascadeStudio API

..........

it will take some time to put massive number of links I hoarded, so I will update this list slowly.

**Comment by Loosetooth** [comment]
Thanks for sharing! :)
I think the most interesting find here is manifoldcad. It's really fast, even in the browser. I'm probably going to look into it the coming days.

Something I found before is called [OpenCascade.js](https://github.com/donalffons/opencascade.js) which is also using webassembly and is faster than OpenJSCAD. For example, there's a web GUI called [replicad](https://replicad.xyz/). Try it out and let me know what you think.

There's also [CascadeStudio](https://zalo.github.io/CascadeStudio/) based on OpenCascade.js.

**Reply by hrgdavor** [reply]
yes, those are also on my bookmark list. I like some aspects of BREP, but currently am not willing to dive deeper into OCCT.
